lib/screens/BookDetailsScreen.dart

```diff
@@ -1,96 +1,345 @@
+                            onPressed: () {
+
+                            },
+                          ),
+                          SizedBox(
+                            width: 10.0,
+                          ),
+                          RaisedButton(
+                            child: new Text('CANCEL'),
+                            color: Colors.deepPurple,
+                            textColor: Colors.white,
+                            onPressed: () {
+                              Navigator.of(context).pop();
+                            },
+                          ),
+                        ],
+                      ),
+                    ),
+                  ],
+                );
+              },
+            ),
+          );
+        }
+    );
+  }
+
+
+  _addToListDialog(BuildContext context) async {
+
+    int selectedRating;
+    String selectedStatus;
+
+    return showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          return AlertDialog(
+            title: Text('Add to list', textAlign: TextAlign.center,),
+            content: StatefulBuilder(
+              builder: (BuildContext context, StateSetter setState) {
+                return Column(
+                  mainAxisSize: MainAxisSize.min,
+                  children: <Widget>[
+                    Text('Status: '),
+                    Container(
+                      padding: EdgeInsets.symmetric(horizontal: 49.0),
+                      child: Row(
+                        children: <Widget>[
+                          DropdownButton(
+                            value: selectedStatus,
+                            items: _possibleStatuses.map((status) {
+                              return DropdownMenuItem(
+                                  child: Text(status), value: status);
+                            }).toList(),
+                            onChanged: (value) {
+                              setState(() => selectedStatus = value);
+                            },
+                          )
+                        ],
+                      ),
+                    ),
+                    Text('Your rating: '),
+                    Container(
+                      padding: EdgeInsets.symmetric(horizontal: 90.0),
+                      child: Row(
+                        children: <Widget>[
+                          DropdownButton(
+                            value: selectedRating,
+                            items: _possibleRatings.map((rating) {
+                              return DropdownMenuItem(
+                                  child: Text(rating.toString()), value: rating);
+                            }).toList(),
+                            onChanged: (value) {
+                              setState(() => selectedRating = value);
+                            },
+                          )
+                        ],
+                      ),
+                    ),
```

Comment:
Repeated code, move creation of Text + Dropdown to function, parametrized by label text and values list.

_createDropdown(String label, List values) {...}

Values will be `_possibleStatuses` and `_possibleRatings.map((r) => r.toString()).toList()`